replica/distributed_loader.cc

```diff
@@ -538,30 +538,16 @@ future<> distributed_loader::init_non_system_keyspaces(distributed<replica::data
         }).get();
 
         const auto& cfg = db.local().get_config();
-        using ks_dirs = std::unordered_multimap<sstring, sstring>;
-
-        ks_dirs dirs;
-
-        parallel_for_each(cfg.data_file_directories(), [&dirs] (sstring directory) {
-            // we want to collect the directories first, so we can get a full set of potential dirs
-            return lister::scan_dir(fs::path{directory}, lister::dir_entry_types::of<directory_entry_type::directory>(),
-                    [&dirs] (fs::path datadir, directory_entry de) {
-                if (!is_system_keyspace(de.name)) {
-                    dirs.emplace(de.name, datadir.native());
-                }
-                return make_ready_future<>();
-            });
-        }).get();
```

Comment:
Don't we loose the ability of detecting orphan keyspace directories? Did we have that ability in the first place?

Reply:
Currently if a directory for a keyspace exists, but the keyspace cannot be found on database, a warning is printed in logs and population continues. Just that. After the patch, such case won't be reported in logs, that's the only difference.

Reply:
I don't think that's an important distinction. And we probably generate that problem in normal operation, if we delete a keyspace and keep a snapshot.